app/controllers/books_controller.rb

```diff
@@ -0,0 +1,52 @@
+class BooksController < ApplicationController
+  def index
+    @books = Book.all
+  end
+
+  def show
+    @book = Book.find(params[:id])
+    # @reviews = @book.reviews
+  end
+
+  def new
+    @book = Book.new
+    @author = Author.new
+    @genres_collection = Genre.all.map do |genre|
+      [genre.genre_name, genre.id]
+    end
+  end
+
+  def create
+    @book = Book.new(book_params )
+    @genres_collection = Genre.all.map do |genre|
+      [genre.genre_name, genre.id]
+    end
+
+
+    @author = Author.find_or_create_by(name: author_params[:author])
+    @book.author = @author
+    #@genre = Genre.find_by()
```

Comment:
Incorrect indentation detected (column 4 instead of 6).<br>Missing space after #.